lib/Compiler.js

```diff
@@ -2,520 +2,522 @@
-		plugin: function(hook, fn) {
-			if(!deprecationReported) {
-				console.warn("webpack: Using compiler.parser is deprecated.\n" +
-					"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.plugin(/* ... */); });\n}); instead. " +
-					"It was called " + new Error().stack.split("\n")[2].trim() + ".");
-				deprecationReported = true;
-			}
-			this.plugin("compilation", function(compilation, data) {
-				data.normalModuleFactory.plugin("parser", function(parser) {
-					parser.plugin(hook, fn);
-				});
-			});
-		}.bind(this),
-		apply: function() {
-			var args = arguments;
-			if(!deprecationReported) {
-				console.warn("webpack: Using compiler.parser is deprecated.\n" +
-					"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.apply(/* ... */); });\n}); instead. " +
-					"It was called " + new Error().stack.split("\n")[2].trim() + ".");
-				deprecationReported = true;
-			}
-			this.plugin("compilation", function(compilation, data) {
-				data.normalModuleFactory.plugin("parser", function(parser) {
-					parser.apply.apply(parser, args);
-				});
-			});
-		}.bind(this)
-	};
-
-	this.options = {};
 }
-module.exports = Compiler;
 
-Compiler.prototype = Object.create(Tapable.prototype);
-Compiler.prototype.constructor = Compiler;
+class Compiler extends Tapable {
+	constructor() {
+		super();
+		this.outputPath = "";
+		this.outputFileSystem = null;
+		this.inputFileSystem = null;
 
-Compiler.Watching = Watching;
-Compiler.prototype.watch = function(watchOptions, handler) {
-	this.fileTimestamps = {};
-	this.contextTimestamps = {};
-	var watching = new Watching(this, watchOptions, handler);
-	return watching;
-};
+		this.recordsInputPath = null;
+		this.recordsOutputPath = null;
+		this.records = {};
 
-Compiler.prototype.run = function(callback) {
-	var self = this;
-	var startTime = Date.now();
+		this.fileTimestamps = {};
+		this.contextTimestamps = {};
 
-	self.applyPluginsAsync("before-run", self, function(err) {
-		if(err) return callback(err);
+		this.resolvers = {
+			normal: null,
+			loader: null,
+			context: null
+		};
+		let deprecationReported = false;
+		this.parser = {
+			plugin: (hook, fn) => {
+				if(!deprecationReported) {
+					console.warn("webpack: Using compiler.parser is deprecated.\n" +
+						"Use compiler.plugin(\"compilation\", function(compilation, data) {\n  data.normalModuleFactory.plugin(\"parser\", function(parser, options) { parser.plugin(/* ... */); });\n}); instead. " +
+						"It was called " + new Error().stack.split("\n")[2].trim() + ".");
+					deprecationReported = true;
+				}
+				this.plugin("compilation", (compilation, data) => {
+					data.normalModuleFactory.plugin("parser", parser => {
+						parser.plugin(hook, fn);
+					});
+				});
+			},
+			apply: () => {
+				const args = arguments;
```

Comment:
`arguments` is not available in an arrow function.

Replace it with a function.

Spread arguments would be nicer, but they are not support in node 4.